crypto/rsa_go.go

```diff
@@ -93,7 +108,35 @@ func (sk *RsaPrivateKey) Raw() ([]byte, error) {
 
 // Equals checks whether this key is equal to another
 func (sk *RsaPrivateKey) Equals(k Key) bool {
-	return KeyEqual(sk, k)
+	// make sure this is an rsa public key
+	other, ok := (k).(*RsaPrivateKey)
+	if !ok {
+		a, err := sk.Raw()
+		if err != nil {
+			return false
+		}
+		b, err := k.Raw()
+		if err != nil {
+			return false
+		}
+		return bytes.Equal(a, b)
+	}
+
+	a := sk.sk
+	b := other.sk
+
+	if a.PublicKey.N.Cmp(b.PublicKey.N) != 0 {
+		return false
+	}
+	if a.PublicKey.E != b.PublicKey.E {
+		return false
+	}
+
+	if a.D.Cmp(b.D) != 0 {
```

Comment:
@Stebalien not sure if we should do this, I would argue if the public key matches, we can assume the private key matches

Reply:
I'd just do it. It shouldn't be expensive.

Reply:
I think we should remove it, openssl only checks the public key, and this check is not constant time, so it could leak information about the private key afict

Reply:
Meh, fine by me either way.

Reply:
Done!